Data/Hashable/Class.hs

```diff
@@ -591,7 +601,7 @@ instance Hashable1 [] where
         step (SP s l) x   = SP (h s x) (l + 1)
 
 instance Hashable B.ByteString where
-    hashWithSalt salt bs = B.inlinePerformIO $
+    hashWithSalt salt bs = unsafeDupablePerformIO $
```

Comment:
@tibbe, I checkout out the current HEAD state of `hashable` (what I will call "current" from here on out), as well as @bgamari's fork (what I will call "proposed"), and ran the benchmarks with both repos using a 4-core, 64-bit Linux desktop with 16 GB of RAM.

A preamble: it should be noted that the benchmarks currently do not compile. You have to make this change first before they will work:

```diff
diff --git a/hashable.cabal b/hashable.cabal
index 2dbf763..80cdf83 100644
--- a/hashable.cabal
+++ b/hashable.cabal
@@ -113,6 +113,7 @@ benchmark benchmarks
     base,
     bytestring,
     criterion >= 1.0,
+    deepseq >= 1.3,
     ghc-prim,
     siphash,
     text
```

Also, it should be noted that the benchmarks still contain uses of `inlinePerformIO` (not that that would affect the benchmark results, but it is worth noting if @bgamari's goal is to purge uses of `inlinePerformIO`).

Here are the results:

* [`bench-results-current.txt`](https://gist.githubusercontent.com/RyanGlScott/d4e30061a54cf05a23a8f43464f5070d/raw/893cad9bff101e1c9e4b837a11344c7a3a3b4da9/bench-results-current.txt)
* [`bench-results-proposed.txt`](https://gist.githubusercontent.com/RyanGlScott/d4e30061a54cf05a23a8f43464f5070d/raw/893cad9bff101e1c9e4b837a11344c7a3a3b4da9/bench-results-proposed.txt)

Happily, I can't spot any significant differences between the results for `current` and `proposed`.